build.gradle.kts

```diff
@@ -0,0 +1,162 @@
+import org.jetbrains.intellij.platform.gradle.IntelliJPlatformType
+import org.jetbrains.intellij.platform.gradle.TestFrameworkType
+import org.jetbrains.intellij.platform.gradle.models.ProductRelease
+
+plugins {
+    id("java") // Java support
+    alias(libs.plugins.gradleIntelliJPlugin) // Gradle IntelliJ Plugin
+    id("jacoco") // Code coverage
+    id("maven-publish")
+    alias(libs.plugins.gradleNexusPublishPlugin)
+}
+
+group = "com.redhat.devtools.intellij"
+version = providers.gradleProperty("projectVersion").get() // Plugin version
+val platformVersion = providers.gradleProperty("ideaVersion").get()
+
+// Set the JVM language level used to build the project.
+java {
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
+    withSourcesJar()
+    withJavadocJar()
+}
+
+repositories {
+    mavenLocal()
+    mavenCentral()
+    intellijPlatform {
+        defaultRepositories()
+    }
+}
+
+dependencies {
+    intellijPlatform {
+        create(IntelliJPlatformType.IntellijIdeaCommunity, platformVersion)
+
+        // Bundled Plugin Dependencies. Uses `platformBundledPlugins` property from the gradle.properties file for bundled IntelliJ Platform plugins.
+        bundledPlugins(providers.gradleProperty("platformBundledPlugins").map { it.split(',') })
+
+        // Plugin Dependencies. Uses `platformPlugins` property from the gradle.properties file for plugin from JetBrains Marketplace.
+        plugins(providers.gradleProperty("platformPlugins").map { it.split(',') })
+
+        // for local plugin -> https://plugins.jetbrains.com/docs/intellij/tools-gradle-intellij-plugin-faq.html#how-to-add-a-dependency-on-a-plugin-available-in-the-file-system
+        //plugins.set(listOf(file("/path/to/plugin/")))
+
+        instrumentationTools()
+
+        testFramework(TestFrameworkType.Platform)
+    }
+
+    implementation(libs.kubernetes.client)
+    implementation(libs.openshift.client)
+    implementation(libs.kubernetes.httpclient.okhttp)
+    implementation(libs.jackson.core)
+    implementation(libs.commons.exec)
+    implementation(libs.commons.lang3)
+    implementation(libs.common.lang)
+
+    // for unit tests
+    testImplementation(libs.junit)
+    testImplementation(libs.assertj.core)
+    testImplementation(libs.mockito.inline)
+    testImplementation(libs.opentest4j) // known issue: https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin-faq.html#missing-opentest4j-dependency-in-test-framework
+
+}
+
+intellijPlatform {
+    buildSearchableOptions = false
```

Comment:
maybe comment why this is set to false so that we know later?

Reply:
when there is no custom settings, jetbrains issues a warning about it when using defaut value.
see https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin-faq.html#how-to-disable-building-the-searchable-options